src/ChangeStream.php

```diff
@@ -67,24 +68,27 @@ public function key()
 
     public function next()
     {
+        $resumable = false;
         try {
             $this->csIt->next();
             if ($this->valid()) {
                 $this->extractResumeToken($this->csIt->current());
                 $this->key++;
             }
         } catch (RuntimeException $e) {
-            $resumable = false;
             if (strpos($e->getMessage(), "not master") !== false) {
                 $resumable = true;
             }
             if ($e->getCode() === self::CURSOR_NOT_FOUND) {
-                $resumable = true;
+               $resumable = true;
             }
-            if ($resumable) {
-                $this->resume();
+            if ($e instanceof ConnectionTimeoutException) {
```

Comment:
Per [`phongo_exception_from_mongoc_domain()`](https://github.com/mongodb/mongo-php-driver/blob/5e016c25edca82da94ebe82bdc3faaaf6d062a4f/php_phongo.c#L138), I see that both general socket errors and server selection failures yield a ConnectionTimeoutException. There are a number of other libmongoc errors in the stream family that will result in a ConnectionException, but those all appear to be related to connection initialization or monitoring. Kudos on catching the more specific case here.